### PR TITLE
[PhotosUI] Move macOS photosui bindings out of photos.cs into photosu…

### PR DESCRIPTION
**Pull request by timrisi**
…i.cs

### CODE REVIEW

**Comment by spouliot** [comment]
```
21:59:03 make[2]: *** [build/mac/mobile-32/Xamarin.Mac.dll] Error 1
21:59:03 make[2]: *** [build/mac/full-32/Xamarin.Mac.dll] Error 1
21:59:04 make[2]: *** [build/mac/full-64/Xamarin.Mac.dll] Error 1
```

File: src/photos.cs [hunk 1]
```diff
@@ -39,6 +39,7 @@ interface PHAdjustmentData : NSCoding, NSSecureCoding {
 	[iOS (8,0)]
 	[TV (10,0)]
 	[Mac (10,13, onlyOn64 : true)]
+	[DisableDefaultCtor] // Crashes mac introspection test
```
Comment (spouliot) [inline src/photos.cs]: You cannot remove it from other profiles, that's a breaking change.

File: src/photosui.cs [hunk 2]
```diff
@@ -87,4 +91,232 @@ interface PHLivePhotoViewDelegate {
 		[Export ("livePhotoView:didEndPlaybackWithStyle:")]
 		void DidEndPlayback (PHLivePhotoView livePhotoView, PHLivePhotoViewPlaybackStyle playbackStyle);
 	}
+#else
```
Comment (spouliot) [inline src/photosui.cs]: could be `#endif`
Everything else is decorated with `[NoiOS][NoTV][NoWatch]`

File: src/photosui.cs [hunk 3]
```diff
@@ -87,4 +91,232 @@ interface PHLivePhotoViewDelegate {
...
+		PHProjectTextElementType TextElementType { get; }
+	}
+
+	[Mac (10,13, onlyOn64 : true)]
+	[NoiOS][NoTV]
+	[Protocol]
+	interface PHProjectExtensionController {
+
+		[Export ("supportedProjectTypes", ArgumentSemantic.Copy)]
+		PHProjectTypeDescription[] SupportedProjectTypes { get; }
```
Comment (dalexsoto) [inline src/photosui.cs]: This should not be a property if is an `@optional` turn this into `GetSupportedProjectTypes()` or maybe is a `@required` one and is just missing `[Abstract]`?

File: src/photos.cs (hunk 1 above)
Comment (timrisi) [inline src/photos.cs]: Whoops

**Pull request merged**